### PR TITLE
fix(Switch, RadioButton, Checkbox): stop propagation on click

### PR DESCRIPTION
**Pull request by marcoskolodny**

### CODE REVIEW

File: src/checkbox.tsx [hunk 1]
```diff
@@ -123,7 +123,13 @@ const Checkbox: React.FC<RenderProps | ChildrenProps> = (props) => {
             role="checkbox"
             aria-checked={value ?? checkedState}
             onKeyDown={disabled ? undefined : handleKeyDown}
-            onClick={disabled ? undefined : handleChange}
+            onClick={(e) => {
+                if (!disabled) {
+                    e.preventDefault();
```
Comment (pladaria) [inline src/checkbox.tsx:128]: this preventDefault shouldn't be needed. There's no default handler in divs

File: src/radio-button.tsx [hunk 2]
```diff
@@ -117,7 +117,13 @@ const RadioButton: React.FC<PropsRender | PropsChildren> = ({
             aria-checked={checked}
             aria-disabled={disabled}
             aria-labelledby={labelId}
-            onClick={disabled ? undefined : () => select(value)}
+            onClick={(e) => {
+                if (!disabled) {
+                    e.preventDefault();
```
Comment (pladaria) [inline src/radio-button.tsx:122]: same here

File: src/checkbox.tsx (hunk 1 above)
Comment (pladaria) [reply src/checkbox.tsx:128]: I'm unsure whether we should use the stopPropagation method on disabled fields or not. I would stopPropagation unconditionally. What's your opinion?

Comment (marcoskolodny) [reply src/checkbox.tsx:128]: It can be an option. I'm not sure what would be the best option from the user experience side. I will ask design what are their thoughts about this.

File: src/__tests__/checkbox-test.tsx [hunk 3]
```diff
@@ -131,3 +131,18 @@ test("with link. Clicking the link doesn't check the checkbox", async () => {
...
+        <ThemeContextProvider theme={makeTheme()}>
+            <button onClick={onPressHandler}>
+                <Checkbox name="checkbox" />
+            </button>
+        </ThemeContextProvider>
+    );
+
+    await userEvent.click(screen.getByRole('checkbox'));
+
+    await waitFor(() => expect(onPressHandler).not.toHaveBeenCalled());
```
Comment (atabel) [inline src/__tests__/checkbox-test.tsx:147]: what's the purpose of the `waitFor` here? the expect will work on the first try, and if it fails it will fail forever. So I'd remove the `waitFor`
```suggestion
    expect(onPressHandler).not.toHaveBeenCalled();
```

File: src/checkbox.tsx (hunk 1 above)
Comment (atabel) [reply src/checkbox.tsx:128]: hmmm... why would you ask design? This is just an implementation detail, isn't it?
I wouldn't add the handler if disabled

File: src/checkbox.tsx [hunk 4]
```diff
@@ -105,7 +105,7 @@ const Checkbox: React.FC<RenderProps | ChildrenProps> = (props) => {
         }
     };
 
-    const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    const handleKeyDown = (event: React.KeyboardEvent) => {
```
Comment (atabel) [inline src/checkbox.tsx:108]: why did you remove the generic?

**Review by atabel — COMMENTED** [review]
Please, review the behavior in Rows with onPress + switch/checkbox/radio. I think you have broken the behavior. Also I'd add some tests for those cases

File: src/checkbox.tsx (hunk 1 above)
Comment (marcoskolodny) [reply src/checkbox.tsx:128]: Because it's related to user experience. Design agrees with Pedro about not propagating the event when it's disabled

Comment (atabel) [reply src/checkbox.tsx:128]: ok for me, then

**Comment by marcoskolodny** [comment]
> Please, review the behavior in Rows with onPress + switch/checkbox/radio. I think you have broken the behavior. Also I'd add some tests for those cases

fixed

**Review by atabel — APPROVED** [review]

**Review by pladaria — APPROVED** [review]
nice, thanks!
